.gitlab/kernel_version_testing/common.yml

```diff
@@ -0,0 +1,285 @@
+    TEST_SETS: "no_tracersuite,only_tracersuite,all_tests"
+  before_script:
+    - set +x
+    - export DD_API_KEY=$(aws ssm get-parameter --region us-east-1 --name ci.datadog-agent.datadog_api_key_org2 --with-decryption --query "Parameter.Value" --out text)
+    - set -x
+    - !reference [.retrieve_linux_go_deps]
+    - !reference [.kernel_matrix_testing_new_profile]
+    - !reference [.write_ssh_key_file]
+  script:
+    - echo "s3://dd-pulumi-state?region=us-east-1&awssdk=v2&profile=$AWS_PROFILE" > $STACK_DIR
+    - pulumi login $(cat $STACK_DIR | tr -d '\n')
+    - inv -e kmt.gen-config --ci --arch=$ARCH --output-file=$VMCONFIG_FILE --sets=$TEST_SETS
+    - inv -e system-probe.start-microvms --provision --vmconfig=$VMCONFIG_FILE $INSTANCE_TYPE_ARG $AMI_ID_ARG --ssh-key-name=$AWS_EC2_SSH_KEY_NAME --ssh-key-path=$AWS_EC2_SSH_KEY_FILE --infra-env=$INFRA_ENV --stack-name=kernel-matrix-testing-${TEST_COMPONENT}-${ARCH}-${CI_PIPELINE_ID} --run-agent
+    - cat $CI_PROJECT_DIR/stack.output
+    - pulumi logout
+  after_script:
+    - export AWS_PROFILE=agent-qa-ci
+    - !reference [.shared_filters_and_queries]
+    - mkdir -p $CI_PROJECT_DIR/libvirt/log/$ARCH $CI_PROJECT_DIR/libvirt/xml/$ARCH
+    - !reference [.get_instance_ip_by_type]
+    - ssh -o StrictHostKeyChecking=no -i $AWS_EC2_SSH_KEY_FILE "ubuntu@$INSTANCE_IP" "sudo virsh list --name | grep -v -E '^$' | xargs -I '{}' sh -c \"sudo virsh dumpxml '{}' > /tmp/ddvm-xml-'{}'.txt\""
+    - scp -o StrictHostKeyChecking=no -i $AWS_EC2_SSH_KEY_FILE "ubuntu@$INSTANCE_IP:/tmp/ddvm-*.log" $CI_PROJECT_DIR/libvirt/log/$ARCH
+    - scp -o StrictHostKeyChecking=no -i $AWS_EC2_SSH_KEY_FILE "ubuntu@$INSTANCE_IP:/tmp/ddvm-xml-*" $CI_PROJECT_DIR/libvirt/xml/$ARCH
+  artifacts:
+    when: always
+    paths:
+      - $CI_PROJECT_DIR/stack.output
+      - $CI_PROJECT_DIR/libvirt
+      - $VMCONFIG_FILE
+
+.kernel_matrix_testing_cleanup:
+  stage: kernel_matrix_testing_cleanup
+  image: 486234852809.dkr.ecr.us-east-1.amazonaws.com/ci/test-infra-definitions/runner$TEST_INFRA_DEFINITIONS_BUILDIMAGES_SUFFIX:$TEST_INFRA_DEFINITIONS_BUILDIMAGES
+  when: always
+  tags: ["arch:amd64"]
+  before_script:
+    - !reference [.kernel_matrix_testing_new_profile]
+  script:
+    - !reference [.shared_filters_and_queries]
+    - aws ec2 describe-instances --filters $FILTER_TEAM $FILTER_MANAGED $FILTER_PIPELINE $FILTER_ARCH $FILTER_INSTANCE_TYPE $FILTER_TEST_COMPONENT --output json --query $QUERY_INSTANCE_IDS | tee -a instance.json
+    - INSTANCE_ID="$(jq -r '.[0][0]' < instance.json)"
+    - echo ${INSTANCE_ID}
+    - |
+      if [[ "${INSTANCE_ID}" != "" ]] && [[ "${INSTANCE_ID}" != "null" ]]; then
+        aws ec2 terminate-instances --instance-ids "${INSTANCE_ID}"
+      fi
+
+# -- Test runners
+.kernel_matrix_testing_run_tests:
+  allow_failure: true
+  variables:
+    AWS_EC2_SSH_KEY_FILE: $CI_PROJECT_DIR/ssh_key
+    RETRY: 2
+    MICROVM_GREP_PATTERN: "$ARCH-$TAG-.*$TEST_SET.*-ddvm"
```

Comment:
This should be removed now. This was used before when the stack output was csv style. We parse json now with `jq` so it is not needed.